src/test/groovy/org/gradle/android/RoomSchemaLocationKspWorkaroundTest.groovy

```diff
@@ -0,0 +1,207 @@
+package org.gradle.android
+
+import org.gradle.testkit.runner.BuildResult
+import org.gradle.testkit.runner.TaskOutcome
+import org.junit.Assume
+import spock.lang.Unroll
+
+import static org.gradle.testkit.runner.TaskOutcome.FROM_CACHE
+import static org.gradle.testkit.runner.TaskOutcome.SUCCESS
+
+@MultiVersionTest
+class RoomSchemaLocationKspWorkaroundTest extends RoomWorkaroundAbstractTest {
+
+    @Unroll
+    def "schemas are generated with Ksp into task-specific directory and are cacheable with kotlin and kapt workers enabled (Android #androidVersion) (Kotlin #kotlinVersion) (Ksp #kspVersion)"() {
+        def kotlinVersionNumber = VersionNumber.parse(kotlinVersion)
+        def kspVersionNumber = VersionNumber.parse(kspVersion)
+        // There are kotlin module version errors when using older versions of kotlin with AGP 7.2.0+ in this configuration
+        Assume.assumeFalse(androidVersion >= VersionNumber.parse("7.2.0-alpha01") && kotlinVersionNumber < VersionNumber.parse("1.5.0"))
+        // Using Kotlin 1.8.0 with Ksp 1.7.22 throws java.lang.NoClassDefFoundError: org/jetbrains/kotlin/gradle/dsl/KotlinJvmOptionsImpl
```

Comment:
Why are we testing those versions together? Kotlin 1.8.0 and KSP 1.7.22? Shouldn't we do Kotlin 1.8.0 with KSP 1.8.0?

Reply:
done and explained in the next comment